source/usr/lib/mcvirt/virtual_machine/virtual_machine.py

```diff
@@ -1218,15 +1218,15 @@ def getVncPort(self):
             self
         )
 
-        if self._getPowerState() is not PowerStates.RUNNING:
-            raise VmAlreadyStoppedException('The VM is not running')
         domain_xml = ET.fromstring(
             self._getLibvirtDomainObject().XMLDesc(
                 libvirt.VIR_DOMAIN_XML_SECURE
             )
         )
 
-        if domain_xml.find('./devices/graphics[@type="vnc"]') is None:
+        if self._getPowerState() is not PowerStates.RUNNING:
```

Comment:
This may as well be preior to the obtaining of the XML, as this will not be required if the VM is powered off. Sorry